app/src/main/java/com/example/cynchen/shoppinglist/MainActivity.java

```diff
@@ -0,0 +1,144 @@
+                //When the cancel button is pressed: nothing is done
+                edititem.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int whichButton) {
+                        // Canceled.
+                    }
+                });
+                edititem.show(); //sets the edititem alertdialog to show
+            }
+        });
+
+        //WHEN LIST VIEW ITEM IS LONGCLICKED: Alert dialog pops up and allows user to choose whether or not they want to delete item
+        listview.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                final String item = (String) (listview.getItemAtPosition(position)); //Gets the string name based off of the position that is clicked
+
+                AlertDialog.Builder deleteitem = new AlertDialog.Builder(
+                        MainActivity.this); //AlertDialog of deleting the item is put in
+                deleteitem.setTitle("Do You Want to Permanently Delete This Item"); //the title is set
+                deleteitem.setMessage("Item you are deleting: " + item); //message shows the item name that they are deleting
+
+                //When Ok is pressed: item is removed from list of values and adapter is updated
+                deleteitem.setPositiveButton("Ok", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int whichButton) {
+                        values.remove(item);
+                        adapter.notifyDataSetChanged();
+                    }
+                });
+
+                //When Cancel is pressed: nothing is done and alertdialog is closed
+                deleteitem.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int whichButton) {
+                        // Canceled.
+                    }
+                });
+                deleteitem.show(); //sets the deleteitem alertdialog to show
+                return true;
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+}
```

Comment:
Fantastic code, in the future we will be having you guys initialize adapters and onClicks in a separate class for cleanliness, but nbd this time. You can also probably get away with one comment per method (methods should be concise enough) that details what the method is doing.